lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -4401,6 +4406,33 @@ swift::TypeBase *SwiftASTContext::ReconstructType(ConstString mangled_typename,
                    *ast_ctx, mangled_typename.GetStringRef())
                    .getPointer();
 
+  // If we couldn't find the type and it's an obj-c type,
+  // walk up the type hierarchy until we find something we can
+  // import, or until we run out of types.
```

Comment:
This comment explains what the code does, but not why. How about:

Objective-C classes sometimes have private subclasses that are invisible to the Swift compiler because they are declared and defined in a .m file. If we can't reconstruct an ObjC type, walk up the type hierarchy until we find something we can import, or until we run out of types.